test/functional/tools/markdown_report_extra_files.xml

```diff
@@ -0,0 +1,45 @@
+<tool id="markdown_report_extra_files" name="markdown_report_extra_files" version="1.0">
+    <command><![CDATA[
+        mkdir '$output_report.extra_files_path';
+        cp '$__tool_directory__/data/1.bed' '$output_report.extra_files_path/test.bed';
+        cp '$__tool_directory__/data/rgWebLogo3_test.jpg' '$output_report.extra_files_path/rgWebLogo3_test.jpg';
+        echo "I am writing this to standard output!";
+        cp '${tool_markdown}' '${output_report}';
+    ]]></command>
+    <configfiles>
+        <configfile name="tool_markdown"><![CDATA[
+# ${title}
+
+Here is an image we just stuck in an extra files path
+
+```galaxy
+history_dataset_as_image(output=output_report, path="rgWebLogo3_test.jpg")
+```
+
+This is a fancy table using fancy table features for an extra files.
+
+```galaxy
+history_dataset_as_table(output=output_report, path="test.bed", title="A Fancy Title!", footer="Here is a small legend.")
+```
+
+The standard out for this tool execution is:
+
+```galaxy
+tool_stdout()
+```
+
+This is my document and I have populated the title from a parameter.
+]]></configfile>
+    </configfiles>
+    <inputs>
+        <param name="title" type="text" label="Title"/>
+    </inputs>
+    <outputs>
+        <!-- <data name="output_directory" format="html"/> -->
+        <data name="output_report" format="tool_markdown"/>
+    </outputs>
+    <tests>
```

Comment:
Add a test here that asserts the title and the stdout?

Reply:
There is I think no new tool evaluation code here - it is all client side and in the APIs the client uses. These tool tests would just be testing existing features - configfiles, cheetah elements, and parameter definitions that are unchanged by this PR. I think the tests will be dead weight in terms of the framework but maybe useful from a demonstration perspective given I do link to these tools in the docs 🤷.